File: examples/java/src/main/java/Parser.java
import org.globalnames.parser.ScientificNameParser;

public class Parser {
    public static void main(String[] args) {
        String jsonStr = ScientificNameParser.instance()
                                             .fromString("Homo sapiens L.")
                                             .renderCompactJson();
        System.out.println(jsonStr);
    }
}

File: examples/R/parser.R
install.packages("jsonlite", repos="http://cran.r-project.org")

require("rJava")
library(jsonlite)

.jinit(classpath=Sys.getenv('GNPARSER_JAR_PATH'))

snp=J('org.globalnames.parser.ScientificNameParser','instance')
result=snp$fromString("Homo sapiens L.")$renderJsonString(FALSE)
jsonResult=fromJSON(result)

stopifnot(jsonResult$parsed)
stopifnot(jsonResult$canonical_name$value == "Homo sapiens")

print("Name is parsed:")
print(result)
